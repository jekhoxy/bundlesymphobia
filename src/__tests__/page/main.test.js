import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import Main from "../../client/page/main";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const initialState = {
  suggestions: [],
};

describe("Test Main page component", () => {
  let store;
  beforeEach(() => {
    store = mockStore(initialState);
  });

  it("Render Main page correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Main />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
