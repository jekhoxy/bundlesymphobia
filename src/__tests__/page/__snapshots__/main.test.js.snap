// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Main page component Render Main page correctly 1`] = `
<div
  className="sc-fzpans dRFVDX"
>
  <div
    className="sc-fzplWN fsfINk"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h3>
      Find the cost of adding a npm package to your bundle
    </h3>
    <div
      className="sc-AxhCb hpcrhp"
    >
      <form
        className="sc-AxhUy lexsms"
        onSubmit={[Function]}
      >
        <input
          className="sc-AxgMl kkFbcO"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Find a package"
          type="text"
          value=""
        />
        <div
          className="sc-AxirZ sc-AxiKw kPBqlq"
          fontSize="inherit"
          rotate={false}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
              fillRule="evenodd"
            />
            <path
              clipRule="evenodd"
              d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </form>
      <div
        className="sc-AxheI deggKy"
        open={false}
      />
    </div>
  </div>
</div>
`;
